<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="dmfind">
<title>dmfind: a network medicine tool for the analysis of omics data • dmfind</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="dmfind: a network medicine tool for the analysis of omics data">
<meta property="og:description" content="dmfind">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dmfind</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/dmfind.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>dmfind: a network medicine tool for the analysis of omics data</h1>
            
      
      
      <div class="d-none name"><code>dmfind.Rmd</code></div>
    </div>

    
    
<hr>
<p><img src="imgs/dmfindLogo.jpg" width="300"></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette describes how to use the R package dmfind, which
implements network diffusion-based analysis of omics data for the
identification of differentially enriched modules. The tool implements
functions for:</p>
<ul>
<li>performing network diffusion (also known as network propagation) of
one or more vectors of gene-weights throughout an interactome of
gene-gene interactions;</li>
<li>defining ranked gene lists, where genes are prioritized according to
the Network Smoothing Index (NSI), which is proportional to gene-weights
and the network proximity of high gene-weights;</li>
<li>assessing the characteristics of the (top) networks defined by the
top ranked genes, based on proximity of high scoring genes, enrichment
in high scoring genes, modularity and number of communities;</li>
<li>obtaining the functional cartography of the top networks;</li>
<li>comparing multiple networks by topology;</li>
<li>visualizing the top networks and the relations between
communities.</li>
</ul>
<p>If you use this package please cite:</p>
<ul>
<li>Bersanelli*, Mosca*, et al., Network diffusion-based analysis of
high-throughput data for the detection of differentially enriched
modules. Sci Rep 6, 34841 (2016). <a href="https://doi.org/10.1038/srep34841" class="external-link uri">https://doi.org/10.1038/srep34841</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>To install this package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="va">devtools</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"emosca-cnr/NPATools"</span>, build_vignettes<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"emosca-cnr/dmfind"</span>, build_vignettes<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="overview-of-the-workflow">Overview of the workflow<a class="anchor" aria-label="anchor" href="#overview-of-the-workflow"></a>
</h3>
<p>The typical steps are the followings:</p>
<ol style="list-style-type: decimal">
<li>Definition of the input matrices;</li>
<li>Calculation of the network-smoothing index;</li>
<li>Definition of the top networks;</li>
<li>Topological analysis.</li>
</ol>
<p>We will consider a toy modular network of 500 “genes” obtained using
<code><a href="https://r.igraph.org/reference/sample_pa.html" class="external-link">igraph::barabasi.game()</a></code>, where a community is enriched in
higher scores compared to the others.</p>
<p><img src="imgs/bg.comm.png" style="width:90.0%"><img src="imgs/boxplot.bg.comm.png" style="width:90.0%"></p>
</div>
</div>
<div class="section level2">
<h2 id="definition-of-the-input-matrices">Definition of the input matrices<a class="anchor" aria-label="anchor" href="#definition-of-the-input-matrices"></a>
</h2>
<p>The analysis requires the following inputs:</p>
<ul>
<li>
<span class="math inline">\(\mathbf{W}\)</span>, an <span class="math inline">\(N \times N\)</span> <em>normalized adjacency
matrix</em>, which represents the interactions between genes, e.g.:</li>
</ul>
<pre><code><span><span class="co">#&gt; 6 x 6 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">#&gt;            G1         G2         G3 G4         G5         G6</span></span>
<span><span class="co">#&gt; G1 .          0.06900656 0.04800154  . .          .         </span></span>
<span><span class="co">#&gt; G2 0.06900656 .          0.04637389  . .          .         </span></span>
<span><span class="co">#&gt; G3 0.04800154 0.04637389 .           . 0.04356068 0.05184758</span></span>
<span><span class="co">#&gt; G4 .          .          .           . .          .         </span></span>
<span><span class="co">#&gt; G5 .          .          0.04356068  . .          0.07001400</span></span>
<span><span class="co">#&gt; G6 .          .          0.05184758  . 0.07001400 .</span></span></code></pre>
<ul>
<li><p><span class="math inline">\(\mathbf{X}_0\)</span>, a <span class="math inline">\(N \times m\)</span> matrix with column vectors
<span class="math inline">\(\mathbf{x_0}\)</span> that contain
gene-related quantities;</p></li>
<li><p>a vector to assign each column of <span class="math inline">\(\mathbf{X}_0\)</span> to any of two classes, only
for differential NSI.</p></li>
</ul>
<p><strong>IMPORTANT</strong>:</p>
<ul>
<li>
<span class="math inline">\(\mathbf{W}\)</span> and <span class="math inline">\(\mathbf{X}_0\)</span> must be defined over the
same identifiers;</li>
<li>the normalization of the adjacency matrix must be done by means of
the function <code>normalize_adj_mat()</code>, which requires a binary
adjacency matrix;</li>
<li>the relevance of the genes corresponding to the rows of <span class="math inline">\(\mathbf{X}_0\)</span> is proportional to their
values, that is, the higher the better; negative values are not
allowed.</li>
</ul>
</div>
<div class="section level2">
<h2 id="the-network-smoothing-index-nsi">The network smoothing index (NSI)<a class="anchor" aria-label="anchor" href="#the-network-smoothing-index-nsi"></a>
</h2>
<p>The network smoothing index (NSI) compares the initial state <span class="math inline">\(\mathbf{x}_0\)</span> with the steady-state <span class="math inline">\(\mathbf{x}_s\)</span> reached after network
diffusion:</p>
<p><span class="math display">\[S_j(\mathbf{x_0}) =
\frac{x_{sj}}{x_{0j}+\epsilon}\]</span> where the parameter <span class="math inline">\(\epsilon\)</span> tunes the relevance of the
initial state. The impact of <span class="math inline">\(\epsilon\)</span> can be assessed by means of the
function <code><a href="../reference/eval_eps.html">eval_eps()</a></code>, which finds an optimal value of <span class="math inline">\(\epsilon\)</span> that takes into account the
network proximity of <span class="math inline">\(\mathbf{x_0}\)</span>
values and <span class="math inline">\(\mathbf{x_s}\)</span> values in
the networks composed of the top <span class="math inline">\(n\)</span>
genes ranked by <span class="math inline">\(S\)</span>. The network
proximity is quantified by <span class="math inline">\(\Omega\)</span>
function <span class="citation">(Bersanelli et al. 2016)</span>:</p>
<p><span class="math display">\[\Omega (n) = \mathbf{y}^T(n) \cdot
\mathbf{A}(n) \cdot \mathbf{y}(n)\]</span> calculated over <span class="math inline">\(\mathbf{y} = \mathbf{x_0}\)</span> or <span class="math inline">\(\mathbf{y} = \mathbf{x_s}\)</span>. Let <span class="math inline">\(O_{\mathbf{x}_0}(\epsilon)\)</span> and <span class="math inline">\(O_{\mathbf{x}_s}(\epsilon)\)</span> be sums of
<span class="math inline">\(\Omega\)</span> values over the top ranking
genes <span class="math inline">\(\{1, 2, ..., n\}\)</span>, calculated
using <span class="math inline">\(\mathbf{x_0}\)</span> and <span class="math inline">\(\mathbf{x_s}\)</span> respectively. The optimal
<span class="math inline">\(\epsilon\)</span> value is found as follows:
<span class="math display">\[\text{arg}\,\max\limits_{\epsilon}\, \Bigg(
\frac{O_{\mathbf{x}_0}(\epsilon)}{\text{max}
(O_{\mathbf{x}_0}(\epsilon))} +
\frac{O_{\mathbf{x}_s}(\epsilon)}{\text{max}(O_{\mathbf{x}_s}(\epsilon)}\Bigg)\]</span></p>
<p>we will see below specific examples for the permutation-adjusted
<span class="math inline">\(S\)</span> and differential <span class="math inline">\(S\)</span>.</p>
<div class="section level3">
<h3 id="permutation-adjusted-s">Permutation-adjusted <span class="math inline">\(S\)</span>;<a class="anchor" aria-label="anchor" href="#permutation-adjusted-s"></a>
</h3>
<p>The Permutation-adjusted <span class="math inline">\(S\)</span> is
defined as: <span class="math display">\[Sp_j = S_j \cdot
-log_{10}(p_j)\]</span> where <span class="math inline">\(p\)</span> is
a statistical assessment of <span class="math inline">\(S\)</span> based
o vertex permutations applied to <span class="math inline">\(\mathbf{X}_0\)</span> and <span class="math inline">\(\mathbf{W}\)</span>. The input matrix <span class="math inline">\(\mathbf{X}_0\)</span> contains one (typically) or
more columns of any type of gene weights (e.g., <span class="math inline">\(−\text{log}_{10}(p)\)</span>):</p>
<pre><code><span><span class="co">#&gt; 6 x 1 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">#&gt;             M</span></span>
<span><span class="co">#&gt; G1 .         </span></span>
<span><span class="co">#&gt; G2 0.02777778</span></span>
<span><span class="co">#&gt; G3 0.01388889</span></span>
<span><span class="co">#&gt; G4 .         </span></span>
<span><span class="co">#&gt; G5 0.19444444</span></span>
<span><span class="co">#&gt; G6 .</span></span></code></pre>
<p>As mentioned above, we must find a value for<span class="math inline">\(\epsilon\)</span>. The default behavior of
<code><a href="../reference/eval_eps.html">eval_eps()</a></code> consists in testing <span class="math inline">\(\epsilon\)</span> values that lie from two orders
of magnitude below the minimum of strictly positive <span class="math inline">\(\mathbf{X}_0\)</span> values to two orders of
magnitude above the maximum of <span class="math inline">\(\mathbf{X}_0\)</span>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>ee <span class="ot">&lt;-</span> <span class="fu">eval_eps</span>(<span class="at">X0 =</span> X0, <span class="at">W =</span> W)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>Generating epsilon based on X0 values...</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># orders of magnitude:  6 </span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># values considered:  24</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>eps<span class="sc">:</span> <span class="fl">0.0001388889</span> <span class="fl">0.0002496515</span> <span class="fl">0.0004487462</span> <span class="fl">0.000806617</span> <span class="fl">0.001449887</span> <span class="fl">0.002606157</span> <span class="fl">0.004684544</span> <span class="fl">0.008420423</span> <span class="fl">0.01513563</span> <span class="fl">0.02720616</span> <span class="fl">0.04890281</span> <span class="fl">0.08790234</span> <span class="fl">0.1580036</span> <span class="fl">0.28401</span> <span class="fl">0.5105054</span> <span class="fl">0.9176286</span> <span class="fl">1.649429</span> <span class="fl">2.964833</span> <span class="fl">5.329259</span> <span class="fl">9.579293</span> <span class="fl">17.21869</span> <span class="fl">30.95043</span> <span class="fl">55.63311</span> <span class="dv">100</span> </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>Performing ND...</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>optimal eps<span class="sc">:</span> <span class="fl">0.5105054</span></span></code></pre></div>
<p>In our case study we obtain a value of about 0.51, which is stored in
the element <code>ee$opt_eps</code>. It is possible to perform a visual
inspection of the effect of <span class="math inline">\(\epsilon\)</span> over <span class="math inline">\(\Omega\)</span> via
<code><a href="../reference/plot_omega_eps.html">plot_omega_eps()</a></code></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_omega_eps.html">plot_omega_eps</a></span><span class="op">(</span><span class="va">ee</span><span class="op">)</span></span></code></pre></div>
<p>which shows the values of <span class="math inline">\(\Omega\)</span>
by rank <img src="imgs/omega_rank1.png" style="width:90.0%"> and the
values of <span class="math inline">\(O_{\mathbf{x}_0}(\epsilon))\)</span> and <span class="math inline">\(O_{\mathbf{x}_s}(\epsilon))\)</span></p>
<p><img src="imgs/omega1.png" style="width:90.0%"></p>
<p>The permutation-based NSI is calculated via
<code><a href="../reference/calc_adjND.html">calc_adjND()</a></code>, where we use the optimal <span class="math inline">\(\epsilon\)</span> value obtained above and
consider 99 permutations between vertices matched by degree:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>resSp <span class="ot">&lt;-</span> <span class="fu">calc_adjND</span>(<span class="at">X0 =</span> X0, <span class="at">W =</span> W, <span class="at">eps =</span> ee<span class="sc">$</span>opt_eps, <span class="at">bin_type =</span> <span class="st">"number"</span>, <span class="at">method =</span> <span class="st">"d"</span>, <span class="at">cut_par =</span> <span class="dv">2</span>, <span class="at">k=</span><span class="dv">99</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>BPPARAM</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>class<span class="sc">:</span> SerialParam</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  bpisup<span class="sc">:</span> <span class="cn">FALSE</span>; bpnworkers<span class="sc">:</span> <span class="dv">1</span>; bptasks<span class="sc">:</span> <span class="dv">0</span>; bpjobname<span class="sc">:</span> BPJOB</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  bplog<span class="sc">:</span> <span class="cn">FALSE</span>; bpthreshold<span class="sc">:</span> INFO; bpstopOnError<span class="sc">:</span> <span class="cn">TRUE</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  bpRNGseed<span class="sc">:</span> ; bptimeout<span class="sc">:</span> <span class="dv">2592000</span>; bpprogressbar<span class="sc">:</span> <span class="cn">FALSE</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  bpexportglobals<span class="sc">:</span> <span class="cn">TRUE</span>; bpforceGC<span class="sc">:</span> <span class="cn">FALSE</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  bplogdir<span class="sc">:</span> <span class="cn">NA</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  bpresultdir<span class="sc">:</span> <span class="cn">NA</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>Permutation type<span class="sc">:</span> degree </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>Total permutations<span class="sc">:</span> <span class="dv">100</span> </span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>Minimum possible FDR<span class="sc">:</span> <span class="fl">0.01</span> </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>cut_par<span class="sc">:</span> <span class="dv">2</span> </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>bin_type<span class="sc">:</span> number </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>vertex sets and bins<span class="sc">:</span> <span class="dv">307</span> <span class="dv">144</span> </span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>Min possible k<span class="sc">:</span> <span class="fl">5.550294e+249</span> </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>ND over X0 permutations</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>ND over W permutations</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>calculation of S</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>calculation of p<span class="sc">-</span>values</span></code></pre></div>
<p>The result is a list that contains input and output quantities.</p>
<pre><code><span><span class="co">#&gt;  [1] "X0"   "Xs"   "eps"  "S"    "pX0"  "pW"   "SpX0" "SpW"  "p"    "Sp"</span></span></code></pre>
<p>The main stuff that one should consider tuning is the permutation
approach. Please read the documentation <code>?calc_adjND()</code>.</p>
</div>
<div class="section level3">
<h3 id="differential-nsi-cases-vs-controls">Differential NSI (cases vs controls)<a class="anchor" aria-label="anchor" href="#differential-nsi-cases-vs-controls"></a>
</h3>
<p>The differential NSI is calculated comparing the NSI of two classes
of subjects, based on their average “profiles” <span class="math inline">\(\bar{\mathbf{x}_{o1}\)</span> and <span class="math inline">\(\bar{\mathbf{x}_{o2}}\)</span>:</p>
<p><span class="math display">\[\Delta S_j = S_j (\bar{\mathbf{x}_{o2}})
- S_j (\bar{\mathbf{x}_{o1}})\]</span> In this case each olumn of the
input matrix <span class="math inline">\(\mathbf{X}_0\)</span> carries
sample-level information (e.g., mutation profiles of patients)</p>
<pre><code><span><span class="co">#&gt; 6 x 10 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">#&gt;                       </span></span>
<span><span class="co">#&gt; G1 . . . . . . 1 . . .</span></span>
<span><span class="co">#&gt; G2 . . . . . . . . . .</span></span>
<span><span class="co">#&gt; G3 1 1 1 1 1 . . . . .</span></span>
<span><span class="co">#&gt; G4 . . . . . . . 1 . .</span></span>
<span><span class="co">#&gt; G5 1 1 1 1 1 . . . . .</span></span>
<span><span class="co">#&gt; G6 1 1 1 1 . . . . . .</span></span></code></pre>
<p>which will be averaged to obtain <span class="math inline">\(\bar{\mathbf{x}_{o1}}\)</span> and <span class="math inline">\(\mathbf{x}_{o2}\)</span> according to a given
partition of subjects in two classes. To find an appropriate <span class="math inline">\(\epsilon\)</span> values for each of the two
classes, we run <code><a href="../reference/eval_eps.html">eval_eps()</a></code> on <span class="math inline">\(\bar{\mathbf{x}_{o1}}\)</span> and <span class="math inline">\(\bar{\mathbf{x}_{o2}}\)</span> :</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>X0ds_means <span class="ot">&lt;-</span> <span class="fu">calc_X0_mean</span>(<span class="at">X0 =</span> X0ds, <span class="at">classes =</span> classes)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>eeds <span class="ot">&lt;-</span> <span class="fu">eval_eps</span>(<span class="at">X0 =</span> X0ds_means, <span class="at">W =</span> W, <span class="at">top =</span> <span class="dv">100</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>enerating epsilon based on X0 values...</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># orders of magnitude:  5 </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># values considered:  20 </span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>eps<span class="sc">:</span> <span class="fl">0.002</span> <span class="fl">0.003534632</span> <span class="fl">0.006246812</span> <span class="fl">0.01104009</span> <span class="fl">0.01951133</span> <span class="fl">0.03448268</span> <span class="fl">0.0609418</span> <span class="fl">0.1077034</span> <span class="fl">0.190346</span> <span class="fl">0.3364015</span> <span class="fl">0.5945277</span> <span class="fl">1.050718</span> <span class="fl">1.856951</span> <span class="fl">3.28182</span> <span class="fl">5.800013</span> <span class="fl">10.25046</span> <span class="fl">18.11579</span> <span class="fl">32.01633</span> <span class="fl">56.58298</span> <span class="dv">100</span> </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>Performing ND...</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>optimal eps<span class="sc">:</span> <span class="fl">1.050718</span> <span class="fl">10.25046</span> </span></code></pre></div>
<p>In our case study we obtain a value of about 1 and 10 for
respectively class 1 and 2. We can now run the analysis by:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>resdS <span class="ot">&lt;-</span> <span class="fu">calc_dS</span>(<span class="at">X0 =</span> X0ds, <span class="at">W =</span> W, <span class="at">classes =</span> classes, <span class="at">eps =</span> eeds<span class="sc">$</span>opt_eps)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>network diffusion</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>calculation of dS</span></code></pre></div>
<p>The result will contain the following quantities:</p>
<pre><code><span><span class="co">#&gt; [1] "X0"  "Xs"  "eps" "S"   "dS"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="visualization-of-the-nsi-values">Visualization of the NSI values<a class="anchor" aria-label="anchor" href="#visualization-of-the-nsi-values"></a>
</h2>
<p>From now on, we will focus only on the results of <span class="math inline">\(Sp\)</span> analysis, because the functions that
will be described below work analogously for <span class="math inline">\(Sp\)</span> and <span class="math inline">\(\Delta
S\)</span>.</p>
<p>The results NSI calculation can be visually explored throug the
function <code><a href="../reference/plot_NSI.html">plot_NSI()</a></code>. In the case of <span class="math inline">\(Sp\)</span> two interesting plots are the
distribution of <span class="math inline">\(S\)</span> values and
p-value (that is “effect size” vs significance):</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NSI.html">plot_NSI</a></span><span class="op">(</span><span class="va">resSp</span>, x<span class="op">=</span><span class="st">"S"</span>, y<span class="op">=</span><span class="st">"p"</span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs/Svsp.png" style="width:90.0%"></p>
<p>and the relation between <span class="math inline">\(S_p\)</span> and
the initial state:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NSI.html">plot_NSI</a></span><span class="op">(</span><span class="va">resSp</span>, x <span class="op">=</span> <span class="st">"X0"</span>, y<span class="op">=</span><span class="st">"Sp"</span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs/SpvsX0.png" style="width:90.0%"></p>
<p>Note that genes with a high initial value gets high <span class="math inline">\(S_p\)</span>. In addition some genes with low
initial value get a high <span class="math inline">\(S_p\)</span>, which
implies a network proximity of these genes to those with high <span class="math inline">\(\mathbf{x}_0\)</span>.</p>
</div>
<div class="section level2">
<h2 id="definition-of-the-top-networks">Definition of the top networks<a class="anchor" aria-label="anchor" href="#definition-of-the-top-networks"></a>
</h2>
<p>The package <code>dmfind</code> provides two types of analysis to
guide the choice of the “top” networks, that is the networks that
contain the interactions between genes with high <span class="math inline">\(\mathbf{X}_0\)</span>:</p>
<ul>
<li>
<strong>network resampling (NR)</strong>, which scores the presence
of genes with high NSI values in network proximity;</li>
<li>
<strong>network enrichment (NE)</strong>, which scores network
enrichment in genes with high <span class="math inline">\(\mathbf{X}_0\)</span>;</li>
<li>
<strong>modularity analysis</strong>, which scores network
modularity</li>
</ul>
<p>These analysis are applied over a number of top genes by <span class="math inline">\(S_p\)</span> or <span class="math inline">\(\Delta
S\)</span>. In human genome-wide analysis, a typical number could be
500.</p>
<div class="section level3">
<h3 id="network-resampling">Network resampling<a class="anchor" aria-label="anchor" href="#network-resampling"></a>
</h3>
<p>Network resampling can be used to asses the presence of significantly
connected modules</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nr_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NR.html">NR</a></span><span class="op">(</span>G <span class="op">=</span> <span class="va">bg</span>, rankedVector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">resSp</span><span class="op">$</span><span class="va">Sp</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>, k <span class="op">=</span> <span class="fl">99</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_NR.html">plot_NR</a></span><span class="op">(</span><span class="va">nr_res</span><span class="op">)</span></span></code></pre></div>
<p>The element <code>nr_res$NRsummary</code> contains the NR results by
rank</p>
<pre><code><span><span class="co">#&gt;        id rank rankingScore    omega p_val</span></span>
<span><span class="co">#&gt; G5     G5    1    1.0000000 0.000000  1.00</span></span>
<span><span class="co">#&gt; G61   G61    2    0.6668873 0.000000  1.00</span></span>
<span><span class="co">#&gt; G286 G286    3    0.6622229 0.441628  1.00</span></span>
<span><span class="co">#&gt; G379 G379    4    0.6576295 1.099258  1.00</span></span>
<span><span class="co">#&gt; G284 G284    5    0.6263519 1.516964  0.52</span></span>
<span><span class="co">#&gt; G455 G455    6    0.5458936 1.878467  0.46</span></span></code></pre>
<p>The visual representation is obtained <code><a href="../reference/plot_NR.html">plot_NR()</a></code>:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NR.html">plot_NR</a></span><span class="op">(</span><span class="va">nr_res</span><span class="op">)</span></span></code></pre></div>
<p>We observe that the real <span class="math inline">\(\Omega\)</span>
values begins to be clearly higher than permuted values after
approximately 20 genes, which indicates the presence of a high scoring
connected gene module:</p>
<p><img src="imgs/NR.png" style="width:90.0%"></p>
</div>
<div class="section level3">
<h3 id="network-enrichment">Network Enrichment<a class="anchor" aria-label="anchor" href="#network-enrichment"></a>
</h3>
<p>The enrichment analysis assess to which extent the top networks are
enriched in high <span class="math inline">\(\mathbf{x}_0\)</span>
values. Depending on how <span class="math inline">\(\mathbf{x}_0\)</span> is defined, this task can be
done by means of Over Representation Analysis (ORA) or Gene Set
Enrichment Analysis (GSEA). In the following example, we explore the
enrichment of the networks formed by the top 50 genes by <span class="math inline">\(S_p\)</span>, using GSEA:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ne_res</span> <span class="op">&lt;-</span> <span class="fu">assess_enrichment</span><span class="op">(</span>G <span class="op">=</span> <span class="va">bg</span>, topList <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">resSp</span><span class="op">$</span><span class="va">Sp</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span><span class="op">)</span>, X0Vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">X0</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">X0</span><span class="op">)</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"gsea"</span>, minNetSize <span class="op">=</span> <span class="fl">5</span>, ranks <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<p>The table <code>ne_res$en_summary</code> cotains the results, which,
as expected, show that the top networks are highly enriched in top
scoring genes:</p>
<pre><code><span><span class="co">#&gt;    rank size        es      nes nperm p_val  adj_p_val q_val       FDRq tags tags_perc list_top</span></span>
<span><span class="co">#&gt; 9    13   13 0.9868501 1.647037   100  0.01 0.01043084    NA 0.01002616    9 0.6923077       11</span></span>
<span><span class="co">#&gt; 20   24   24 0.9624399 1.645404   100  0.01 0.01043084    NA 0.01002616   11 0.4583333       14</span></span>
<span><span class="co">#&gt; 8    12   12 0.9886723 1.640034   100  0.01 0.01043084    NA 0.01002616    8 0.6666667        9</span></span>
<span><span class="co">#&gt; 14   18   18 0.9705192 1.638376   100  0.01 0.01043084    NA 0.01002616   10 0.5555556       14</span></span>
<span><span class="co">#&gt; 19   23   23 0.9586542 1.635739   100  0.01 0.01043084    NA 0.01002616   10 0.4347826       14</span></span>
<span><span class="co">#&gt; 23   27   27 0.9568215 1.633151   100  0.01 0.01043084    NA 0.01002616   13 0.4814815       15</span></span>
<span><span class="co">#&gt;    list_top_perc lead_edge</span></span>
<span><span class="co">#&gt; 9     0.02439024 0.6954689</span></span>
<span><span class="co">#&gt; 20    0.03104213 0.4690671</span></span>
<span><span class="co">#&gt; 8     0.01995565 0.6712225</span></span>
<span><span class="co">#&gt; 14    0.03104213 0.5606877</span></span>
<span><span class="co">#&gt; 19    0.03104213 0.4439252</span></span>
<span><span class="co">#&gt; 23    0.03325942 0.4951083</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="comparison-of-nr-and-ne">Comparison of NR and NE<a class="anchor" aria-label="anchor" href="#comparison-of-nr-and-ne"></a>
</h3>
<p>The results of NR and NE can be combined to find possible networks
that are optimal based on the two analyses. In our case study we found
that the network composed of the top 24 genes is associated with the
most significant p-values from NR and NE:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>nr_ne_cmp <span class="ot">&lt;-</span> <span class="fu">cmp_top_net_scores</span>(<span class="at">NRRes =</span> nr_res, <span class="at">netEnrRes =</span> ae_res)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>Smallest highest scoring network at rank<span class="sc">:</span>  <span class="dv">24</span> </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>Overlapping points detected, adding some jitter</span></code></pre></div>
<p>The result is a table the reports the statistics rank by rank:</p>
<pre><code><span><span class="co">#&gt;   rank p_val_NR   p_val_NE    score</span></span>
<span><span class="co">#&gt; 1    5     0.52 0.02105263 1.219172</span></span>
<span><span class="co">#&gt; 2    6     0.46 0.01020408 1.524909</span></span>
<span><span class="co">#&gt; 3    7     0.32 0.01020408 1.658422</span></span>
<span><span class="co">#&gt; 4    8     0.17 0.01020408 1.894081</span></span>
<span><span class="co">#&gt; 5    9     0.17 0.01010101 1.897891</span></span>
<span><span class="co">#&gt; 6   10     0.15 0.01000000 1.948715</span></span></code></pre>
<p>The function also produces the figure <code>NR_vs_NE.png</code>.</p>
<p><img src="imgs/NR_vs_NE.png" style="width:100.0%"></p>
</div>
<div class="section level3">
<h3 id="modularity-assessment">Modularity assessment<a class="anchor" aria-label="anchor" href="#modularity-assessment"></a>
</h3>
<p>The function <code>assess_modularity()</code> quantifies the
community structure and the modularity among the networks composed of
the top ranking genes. This is another piece of information that can be
useful to select the top networks. In this example, we study the
modularity between the top 50 genes sorted by <span class="math inline">\(Sp\)</span>:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assess_mod_res</span> <span class="op">&lt;-</span> <span class="fu">assess_modularity</span><span class="op">(</span>G <span class="op">=</span> <span class="va">bg</span>, vertices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">resSp</span><span class="op">$</span><span class="va">Sp</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">resSp</span><span class="op">$</span><span class="va">Sp</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span><span class="op">]</span>, ranks <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<p>The output is a table with the trend of the modularity and number of
communities at various ranks between the top genes considered.</p>
<pre><code><span><span class="co">#&gt;   rank n_vertex n_edge  algorithm modularity n_community n_vertex_max n_edge_max algorithm_max</span></span>
<span><span class="co">#&gt; 1    1        0      0 fastgreedy        NaN        -Inf            1          0    fastgreedy</span></span>
<span><span class="co">#&gt; 2    2        0      0 fastgreedy        NaN        -Inf            1          0    fastgreedy</span></span>
<span><span class="co">#&gt; 3    3        2      1   multilev  0.0000000           1            2          1      multilev</span></span>
<span><span class="co">#&gt; 4    4        4      2 fastgreedy  0.5000000           2            2          1      multilev</span></span>
<span><span class="co">#&gt; 5    5        5      3 fastgreedy  0.4444444           2            3          2      multilev</span></span>
<span><span class="co">#&gt; 6    6        6      4 fastgreedy  0.4062500           3            4          3    fastgreedy</span></span>
<span><span class="co">#&gt;   modularity_max n_community_max</span></span>
<span><span class="co">#&gt; 1            NaN               1</span></span>
<span><span class="co">#&gt; 2            NaN               1</span></span>
<span><span class="co">#&gt; 3      0.0000000               1</span></span>
<span><span class="co">#&gt; 4      0.0000000               1</span></span>
<span><span class="co">#&gt; 5      0.0000000               1</span></span>
<span><span class="co">#&gt; 6      0.1666667               2</span></span></code></pre>
<p>These results can be plotted through
<code>plot_modu_trend()</code>:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_modu_trend</span><span class="op">(</span><span class="va">assess_mod_res</span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs/mod_trend.png" style="width:90.0%"></p>
</div>
<div class="section level3">
<h3 id="extraction-of-the-top-networks">Extraction of the top networks<a class="anchor" aria-label="anchor" href="#extraction-of-the-top-networks"></a>
</h3>
<p>Once the rank that is associated to the top networks that one
considers optimal, these can be extracted by means of
<code><a href="../reference/extract_module.html">extract_module()</a></code>. Here we consider the top 120 genes by
<span class="math inline">\(S_p\)</span>:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_nets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_module.html">extract_module</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">bg</span>, selectedVertices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">resSp</span><span class="op">$</span><span class="va">Sp</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">resSp</span><span class="op">$</span><span class="va">Sp</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span><span class="op">]</span>, NSIRes <span class="op">=</span> <span class="va">resSp</span>, minSubnetSize <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; IGRAPH 9e6700b UN-- 20 21 -- Barabasi graph</span></span>
<span><span class="co">#&gt; + attr: name (g/c), power (g/n), m (g/n), zero.appeal (g/n), algorithm (g/c), name (v/c),</span></span>
<span><span class="co">#&gt; | comm_id (v/n), X0 (v/n), Sp (v/n), S (v/n), p (v/n), subnetId (v/n), subnetSize (v/n)</span></span>
<span><span class="co">#&gt; + edges from 9e6700b (vertex names):</span></span>
<span><span class="co">#&gt;  [1] G3  --G5   G3  --G6   G5  --G6   G6  --G29  G5  --G58  G6  --G150 G150--G230 G58 --G284</span></span>
<span><span class="co">#&gt;  [9] G61 --G284 G61 --G286 G5  --G298 G5  --G379 G3  --G388 G286--G398 G388--G413 G5  --G417</span></span>
<span><span class="co">#&gt; [17] G286--G417 G29 --G445 G152--G455 G286--G455 G388--G476</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="topological-charcaterization-of-top-networks">Topological charcaterization of top networks<a class="anchor" aria-label="anchor" href="#topological-charcaterization-of-top-networks"></a>
</h2>
<div class="section level3">
<h3 id="community-structure">Community structure<a class="anchor" aria-label="anchor" href="#community-structure"></a>
</h3>
<p>The function <code>find_communities()</code> runs a series of
algorithms to detect the presence of communities:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_comm</span> <span class="op">&lt;-</span> <span class="fu">find_communities</span><span class="op">(</span><span class="va">top_nets</span><span class="op">)</span></span></code></pre></div>
<p>The element <code>res_comm$info</code> contains the modularity score
<span class="math inline">\(Q\)</span> and the number of
communities:</p>
<pre><code><span><span class="co">#&gt;    algorithm modularity n</span></span>
<span><span class="co">#&gt; 1 fastgreedy  0.5079365 5</span></span>
<span><span class="co">#&gt; 2   multilev  0.4682540 5</span></span></code></pre>
<p>In this case the fastgreedy algorithm found a better partition with
the same number of community, and therefore we assign its results to the
igraph object containing the top networks:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_nets</span> <span class="op">&lt;-</span> <span class="fu">assign_communities</span><span class="op">(</span><span class="va">top_nets</span>, <span class="va">res_comm</span><span class="op">$</span><span class="va">comm</span><span class="op">$</span><span class="va">fastgreedy</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="functional-cartography">Functional cartography<a class="anchor" aria-label="anchor" href="#functional-cartography"></a>
</h3>
<p>The function <code>functional_cartography()</code> performs the
functional cartography analysis <span class="citation">(Guimera and
Nunes Amaral 2005)</span>, where each gene is scored by its
participation coefficient <span class="math inline">\(P\)</span> and the
within module degree score, which is a z-score <span class="citation">(Guimera and Nunes Amaral 2005)</span> or a probability
of interaction <span class="citation">(Joyce et al. 2010)</span>. These
values are used to classify genes in 7 classes:</p>
<ul>
<li><p>ultra peripheral nodes (R1): nodes with all their links within
their module;</p></li>
<li><p>peripheral nodes (R2): nodes with most links within their
module;</p></li>
<li><p>non-hubs connector nodes (R3): nodes with many links to other
modules;</p></li>
<li><p>non-hubs kinless nodes (R4): nodes with links homogeneously
distributed among all modules.</p></li>
<li><p>provincial nodes (R5): hub nodes with the vast majority of links
within their module (pc &lt;= 0.3);</p></li>
<li><p>connector hubs (R6): hubs with many links to most of the other
modules (0.30 &lt; pc &lt;= 0.75);</p></li>
<li><p>kinless hubs (R7): hubs with links homogeneously distributed
among all modules (pc &gt; 0.75).</p></li>
</ul>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_nets</span> <span class="op">&lt;-</span> <span class="fu">functional_cartography</span><span class="op">(</span><span class="va">top_nets</span><span class="op">)</span></span></code></pre></div>
<p>The output is the igraph object given in input with the addition of
vertex attributes <code>R</code>, <code>P</code>, and
<code>wmd_score</code>.</p>
<pre><code><span><span class="co">#&gt; IGRAPH 9e6700b UN-- 20 21 -- Barabasi graph</span></span>
<span><span class="co">#&gt; + attr: name (g/c), power (g/n), m (g/n), zero.appeal (g/n), algorithm (g/c), name (v/c),</span></span>
<span><span class="co">#&gt; | comm_id (v/n), X0 (v/n), Sp (v/n), S (v/n), p (v/n), subnetId (v/n), subnetSize (v/n),</span></span>
<span><span class="co">#&gt; | R (v/c), P (v/n), wmd_score (v/n)</span></span>
<span><span class="co">#&gt; + edges from 9e6700b (vertex names):</span></span>
<span><span class="co">#&gt;  [1] G3  --G5   G3  --G6   G5  --G6   G6  --G29  G5  --G58  G6  --G150 G150--G230 G58 --G284</span></span>
<span><span class="co">#&gt;  [9] G61 --G284 G61 --G286 G5  --G298 G5  --G379 G3  --G388 G286--G398 G388--G413 G5  --G417</span></span>
<span><span class="co">#&gt; [17] G286--G417 G29 --G445 G152--G455 G286--G455 G388--G476</span></span></code></pre>
<p>A dedicated function exist to obtain the functional cartography:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_fc</span><span class="op">(</span><span class="va">top_nets</span>, labelBy <span class="op">=</span> <span class="st">"name"</span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs/fc.png" style="width:90.0%"></p>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>The function <code>plot_network()</code> provides a means to produce
network visualizations. Please read its documentation
(<code>?plot_network</code>) for further details. For example, here we
plot the top network producing a layout that takes into account the
community structure and color by community:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lo_tn</span> <span class="op">&lt;-</span> <span class="fu">plot_network</span><span class="op">(</span><span class="va">top_nets</span>, vertex.size<span class="op">=</span><span class="fl">5</span>, colorQuant <span class="op">=</span> <span class="cn">T</span>, colorBy <span class="op">=</span> <span class="st">"comm_id"</span>, community <span class="op">=</span> <span class="va">res_comm</span><span class="op">$</span><span class="va">comm</span><span class="op">$</span><span class="va">fastgreedy</span>, pal<span class="op">=</span><span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">alphabet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">res_comm</span><span class="op">$</span><span class="va">comm</span><span class="op">$</span><span class="va">fastgreedy</span><span class="op">$</span><span class="va">membership</span><span class="op">)</span><span class="op">)</span>, labelBy <span class="op">=</span> <span class="st">"name"</span>,  vertex.label.degree<span class="op">=</span><span class="va">pi</span><span class="op">/</span><span class="fl">2</span>, vertex.label.dist<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs/top_net_comm.png" style="width:90.0%"></p>
<p>while in this second example we color by <span class="math inline">\(\mathbf{X_0}\)</span></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_network</span><span class="op">(</span><span class="va">top_nets</span>, vertex.size<span class="op">=</span><span class="fl">5</span>, colorQuant <span class="op">=</span> <span class="cn">T</span>, colorBy <span class="op">=</span> <span class="st">"X0"</span>, pal<span class="op">=</span><span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/brewer.html" class="external-link">brewer.purples</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, lo <span class="op">=</span> <span class="va">lo_tn</span>, labelBy <span class="op">=</span> <span class="st">"name"</span>, vertex.label.degree<span class="op">=</span><span class="va">pi</span><span class="op">/</span><span class="fl">2</span>, vertex.label.dist<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs/top_net_X0.png" style="width:90.0%"> and lastly by
<span class="math inline">\(S_p\)</span>:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_network</span><span class="op">(</span><span class="va">top_nets</span>, vertex.size<span class="op">=</span><span class="fl">5</span>, colorQuant <span class="op">=</span> <span class="cn">T</span>, colorBy <span class="op">=</span> <span class="st">"Sp"</span>, pal<span class="op">=</span><span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/brewer.html" class="external-link">brewer.purples</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, lo <span class="op">=</span> <span class="va">lo_tn</span>, labelBy <span class="op">=</span> <span class="st">"name"</span>, vertex.label.degree<span class="op">=</span><span class="va">pi</span><span class="op">/</span><span class="fl">2</span>, vertex.label.dist<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="imgs/top_net_Sp.png" style="width:90.0%"></p>
</div>
<div class="section level2">
<h2 id="interactome-derived-from-string-v12">Interactome derived from STRING v12<a class="anchor" aria-label="anchor" href="#interactome-derived-from-string-v12"></a>
</h2>
<p>This package contains an interactome derived from STRING db v12.0
<span class="citation">(Szklarczyk et al. 2022)</span>, consisting of
17288 genes and 174962 interactions. Only interactions with confidence
&gt;=700 and the top 3 interactions with confidence &gt;=400 &amp; &lt;
700 were considered. Confidence score was calculated without text mining
evidence, as described at the URL <a href="https://string-db.org" class="external-link uri">https://string-db.org</a>. Original protein identifiers were
mapped to Entrez Gene identifiers using mappings provided by STRING (<a href="https://string-db.org" class="external-link uri">https://string-db.org</a>) and
NCBI (<a href="https://ftp.ncbi.nlm.nih.gov" class="external-link uri">https://ftp.ncbi.nlm.nih.gov</a>, download date 2023-09-19).
The interactome is available as an igraph object:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dmfind</span><span class="op">)</span></span>
<span><span class="va">string.v12.Entrez.ntm.700.400.k3</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; IGRAPH 6f011b4 UN-- 17288 174962 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), symbol (v/c), neighborhood (e/n), neighborhood_transferred (e/n),</span></span>
<span><span class="co">#&gt; | fusion (e/n), cooccurence (e/n), homology (e/n), coexpression (e/n),</span></span>
<span><span class="co">#&gt; | coexpression_transferred (e/n), experiments (e/n), experiments_transferred (e/n),</span></span>
<span><span class="co">#&gt; | database (e/n), database_transferred (e/n), combined_score (e/n)</span></span>
<span><span class="co">#&gt; + edges from 6f011b4 (vertex names):</span></span>
<span><span class="co">#&gt;  [1] 1066--1571   1066--4353   1066--9      1066--978    1066--7365   1066--2990   1066--8824  </span></span>
<span><span class="co">#&gt;  [8] 1066--7363   1066--1576   1066--10     1066--79799  1544--1548   1544--1571   1544--7498  </span></span>
<span><span class="co">#&gt; [15] 1544--9      1544--1558   1544--56603  1544--1551   1544--8854   1544--7364   1544--316   </span></span>
<span><span class="co">#&gt; [22] 1544--412    1544--220    1544--8644   1544--1559   1544--1562   1544--1555   1544--216   </span></span>
<span><span class="co">#&gt; [29] 1544--1579   1544--3284   1544--6820   1544--29785  1544--284541 1544--340665 1544--1586  </span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span></code></pre>
<p>The matrix <span class="math inline">\(W\)</span> can be derived as
follows:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">A</span> <span class="op">&lt;-</span> <span class="fu">get.adjacency</span><span class="op">(</span><span class="va">string.v12.Entrez.ntm.700.400.k3</span>, type <span class="op">=</span> <span class="st">"both"</span><span class="op">)</span></span>
<span><span class="va">W</span> <span class="op">&lt;-</span> <span class="fu">normalize_adj_mat</span><span class="op">(</span><span class="va">A</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Bersanelli2016" class="csl-entry">
Bersanelli, Matteo, Ettore Mosca, Daniel Remondini, Gastone Castellani,
and Luciano Milanesi. 2016. <span>“Network Diffusion-Based Analysis of
High-Throughput Data for the Detection of Differentially Enriched
Modules.”</span> <em>Scientific Reports</em> 6 (1): 34841.
</div>
<div id="ref-Guimera2005" class="csl-entry">
Guimera, Roger, and Luı́s A Nunes Amaral. 2005. <span>“Functional
Cartography of Complex Metabolic Networks.”</span> <em>Nature</em> 433
(7028): 895–900.
</div>
<div id="ref-Joyce2010" class="csl-entry">
Joyce, Karen E, Paul J Laurienti, Jonathan H Burdette, and Satoru
Hayasaka. 2010. <span>“A New Measure of Centrality for Brain
Networks.”</span> <em>PloS One</em> 5 (8): e12200.
</div>
<div id="ref-Szklarczyk2022" class="csl-entry">
Szklarczyk, Damian, Rebecca Kirsch, Mikaela Koutrouli, Katerina Nastou,
Farrokh Mehryary, Radja Hachilif, Annika L Gable, et al. 2022.
<span>“<span class="nocase">The STRING database in 2023: protein–protein
association networks and functional enrichment analyses for any
sequenced genome of interest</span>.”</span> <em>Nucleic Acids
Research</em> 51 (D1): D638–46. <a href="https://doi.org/10.1093/nar/gkac1000" class="external-link">https://doi.org/10.1093/nar/gkac1000</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ettore Mosca, Valentina Nale.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
